src/probability/dirac.jl

```diff
@@ -2,21 +2,17 @@
 export Dirac
 
 struct Dirac{X} <: AbstractMeasure
-    supp :: X
+    x::X
 end
 
-Dirac(x::X) = Dirac{X}(x)
 
-sampletype(μ::Dirac{X}) = X
+sampletype(μ::Dirac{X}) where {X} = X
 
 function (μ::Dirac{X})(s) where {X}
-    μ.supp ∈ s && return 1
+    μ.x ∈ s && return 1
     return 0
 end
+isprimitive(::Dirac) = true
 
-function logdensity(μ::Dirac{X}, x::X)
```

Comment:
not a density, rather a primitive

Reply:
Good catch, thanks! It will have a density wrt a discrete measure over a large support, but the for I use here assumes there's a base measure, which is not right.